src/lib/cdev/nuttx/cdev_platform.cpp

```diff
@@ -118,6 +135,11 @@ cdev_write(file_t *filp, const char *buffer, size_t buflen)
 static off_t
 cdev_seek(file_t *filp, off_t offset, int whence)
 {
+	if ((filp->f_inode->i_flags & FSNODEFLAG_DELETED) != 0) {
+		errno = ENODEV;
```

Comment:
@dagar, I'm only not sure about this one. seek can return negative values in valid cases, so the only way is use errno. Maybe this linux-style approach is suitable for all these functions, but as I see usually it is not used in NuttX.

Reply:
This seems like the right thing to do. 

https://pubs.opengroup.org/onlinepubs/009604599/functions/lseek.html

What valid case returning a negative value are you thinking of?

Reply:
Oh, my bad, for some reason I thought that actual value of offset is returned, so on moving backward it would be negative. But it is offset from the beginning of the file, so any negative value on error is ok. Thanks. I'll fix it now.

Reply:
No problem, maybe worth a quick pass to make them all conform to the open group base specification?

Reply:
I thought about this. But I took a look how it is done in different NuttX drivers and in all places I looked error is returned as value. I think in this case it would be better to follow NuttX way rather than specification, wouldn't it?

Reply:
Also PX4 drivers already follow NuttX way, for example IridiumSBD
https://github.com/PX4/PX4-Autopilot/blob/c9b5e488f50561be0aeb9a50d59d09514e3dba0a/src/drivers/telemetry/iridiumsbd/IridiumSBD.cpp#L705
so it is not a good thing to force mix different user logic.

Reply:
That's likely an acceptable choice until we have the time or patience to do the software archeology. At the moment I don't know if it was an intentional choice (efficiency or laziness) in those different NuttX drivers.

The most important thing is to first eliminate any potential hard faults.